manim/utils/bezier.py

```diff
@@ -369,3 +370,108 @@ def get_quadratic_approximation_of_cubic(a0, h0, h1, a1):
 
 def is_closed(points: typing.Tuple[np.ndarray, np.ndarray]) -> bool:
     return np.allclose(points[0], points[-1])
+
+
+def bezier_params_from_point(
+    point: typing.Iterable[typing.Union[float, int]],
+    control_points: typing.Iterable[typing.Iterable[typing.Union[float, int]]],
+    round_to: typing.Optional[typing.Union[float, int]] = 1e-6,
+) -> typing.List[float]:
+    """Obtains the parameter corresponding to a given point
+    for a given bezier curve.
+
+    Parameters
+    ----------
+    point
+        The Cartesian Coordinates of the point whose parameter
+        should be obtained.
+    control_points
+        The Cartesian Coordinates of the ordered control
+        points of the bezier curve on which the point may
+        or may not lie.
+    round_to
+        A float whose number of decimal places all values
+        such as coordinates of points will be rounded.
+
+    Returns
+    -------
+        list[float]
+            List containing possible parameters for the given point on
+            the given bezier curve.
+            This usually only contains one or zero elements, but if the
+            point is, say, at the beginning/end of a closed loop, may return
+            a list with more than 1 value, corresponding to the beginning and
+            end etc. of the loop.
+
+    Raises
+    ------
+    :class:`ValueError`
+        When ``point`` and the control points have different shapes.
+    """
+    if not all(np.shape(point) == np.shape(c_p) for c_p in control_points):
+        raise ValueError(
+            f"Point {point} and Control Points {control_points} have different shapes."
+        )
+    control_points = np.array(control_points)
+    n = len(control_points) - 1
+
+    roots = []
+    for dim, coord in enumerate(point):
+        control_coords = c_c = control_points[:, dim]
+        terms = []
+        for term_power in range(n, -1, -1):
+            outercoeff = choose(n, term_power)
+            term = []
+            negative = 1
+            for subterm_num in range(term_power, -1, -1):
+                innercoeff = choose(term_power, subterm_num) * negative
+                subterm = (
+                    innercoeff * c_c[subterm_num]
+                    if term_power > 0
+                    else ((innercoeff * c_c[subterm_num]) - coord)
+                )
+                term.append(subterm)
+                negative = -1 if negative > 0 else 1
```

Comment:
```suggestion
                negative *= -1
```
(or `sign *= -1` if it's renamed to that)

Reply:
Unfortunately I won't be able to do that. The pattern works by alternating between positive and negative signs, where the first subterm, no matter if its odd or even is always positive.